src/widget/hotcuedrag.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <QBuffer>
+#include <QMimeData>
+#include <QString>
+
+#include "track/cue.h"
+#include "track/trackid.h"
+
+const QString kDragMimeType = QStringLiteral("hotcueDragInfo");
+
+class TrackId;
+class QDragEnterEvent;
+class QDropEvent;
+
+namespace mixxx {
+
+namespace hotcuedrag {
+
+struct HotcueDragInfo {
+    constexpr HotcueDragInfo() {};
+    constexpr HotcueDragInfo(TrackId id, int cue)
+            : trackId(id),
+              hotcue(cue) {};
+
+    static HotcueDragInfo fromByteArray(const QByteArray& bytes) {
+        QDataStream stream(bytes);
+        TrackId trackId;
+        int hotcue;
+        stream >> trackId >> hotcue;
+        return HotcueDragInfo(trackId, hotcue);
+    };
+
+    QByteArray toByteArray() const {
+        QByteArray bytes;
+        QDataStream dataStream(&bytes, QIODevice::WriteOnly);
+        dataStream << trackId << hotcue;
+        return bytes;
+    };
+
+    bool isValid() const {
+        return trackId.isValid() && hotcue != Cue::kNoHotCue;
+    }
+
+    TrackId trackId = TrackId();
+    int hotcue = Cue::kNoHotCue;
+};
+
+/// Check if the event is a valid hotcue drag or drop event.
+/// Event must be a QDragEnterEvent or a QDropEvent.
+/// In case of QDropEvent, the HotcueDragInfo is extracted and the pointer
+/// data is used by the caller (WHotcueButton) to swap hotuces.
+template<typename T>
+bool isValidHotcueDragOrDropEvent(T* pEvent,
+        const QString& group,
+        QObject* pTarget = nullptr,
+        int ignoreHotcueIndex = Cue::kNoHotCue,
+        HotcueDragInfo* pDragData = nullptr);
```

Comment:
this (fortunately) can't be called anymore, so remove it from the header. 
```suggestion
```

Reply:
Okay, so I didn't get completely how namespaces work.
It's not a class so it can be defined in cpp only. Nice! Now everything makes sense.

Reply:
well, its not really whether its a class or not, its about whether its accessible from other translation units. Since only the wrappers are accessible, only they need to be declared in the header.

Reply:
> its not really whether its a class or not,  its about whether its accessible from other translation units

yeah, I meant to say member function (not inline/namespace) which needs a declaration.
Thanks!